packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/TransformHelper.java

```diff
@@ -130,6 +149,75 @@ public static void processTransform(
     }
   }
 
+  public static Matrix tryProcessTransformBySkiaMatrix(ReadableArray transforms,
+                                                       float viewWidth,
+                                                       float viewHeight,
+                                                       ReadableArray transformOrigin) {
+    if (transforms.size() == 16 && transforms.getType(0) == ReadableType.Number) {
+      return null;
+    } else {
+      // Check for unsupported types.
+      for (int transformIdx = 0, size = transforms.size(); transformIdx < size; transformIdx++) {
+        ReadableMap transform = transforms.getMap(transformIdx);
+        String transformType = transform.keySetIterator().nextKey();
+
+        if ("matrix".equals(transformType) || "perspective".equals(transformType)
+          || "rotateX".equals(transformType) || "rotateY".equals(transformType)) {
+          return null;
+        } else if ("translate".equals(transformType)) {
+          ReadableArray value = transform.getArray(transformType);
+          if (value.size() > 2 && value.getDouble(2) != 0d) {
+            return null;
+          }
+        }
+      }
+
+      Matrix matrix = new Matrix();
+      float[] offsets = getTranslateForTransformOrigin(viewWidth, viewHeight, transformOrigin);
+      float originX = viewWidth / 2, originY = viewHeight / 2;
+      if (offsets != null) {
+        originX += offsets[0];
+        originY += offsets[1];
+      }
```

Comment:
I think we should account for this by doing a transform as a first step and then the reverse transform as the last step, similar to `processTransform`

Reply:
done